lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -1019,7 +1033,13 @@ def create_servers(action_handler, specs_and_options, parallelizer, &block)
             instance.source_dest_check = machine_options[:source_dest_check] if machine_options.has_key?(:source_dest_check)
             converge_tags(instance, machine_options[:aws_tags], action_handler)
             machine_spec.reference['key_name'] = bootstrap_options[:key_name] if bootstrap_options[:key_name]
-            %w(is_windows ssh_username sudo use_private_ip_for_ssh ssh_gateway).each do |key|
+            # TODO 2.0 We no longer support `use_private_ip_for_ssh`, only `transport_address_location`
+            if machine_options[:use_private_ip_for_ssh]
+              Chef::Log.warn("The machine_option ':use_private_ip_for_ssh' has been deprecated, use ':transport_address_location'")
```

Comment:
Do you have a good pattern example of a way to only warn once per run?  Just a class variable?